### ISSUE TITLE
"First Class Callable Syntax" optional?

### ISSUE THREAD

**Issue by L1Q** [php/doc-en#1427]
From manual page: https://php.net/functions.first_class_callable_syntax

---

Is it possible to have this feature disabled? The docs never mention it, but I encountered a server running PHP 8.1.2 (according to ``echo phpinfo()``)that does not recognize ``functionname(...)`` syntax and throws "ArgumentCountError".

Is there a setting I can check in ``phpinfo()`` to know for sure this is intended? E.g. the website hosting provider disabled the feature or is using a build of PHP without it?

Most importantly for the docs, any info about how to figure this out before doing a major refactor and finding out the thing would not run in production would be appreciated.

**Comment by damianwadley** [comment]
It cannot be disabled.

If you're having problems with some code then please post the code.

**Comment by L1Q** [comment]
Thanks for quick response, here is my MRE:
```php
<?php

namespace Test;

function greeter(string $name) {
  echo "Hello, ${name}!";
}

$mycallable = greeter(...);

$mycallable(PHP_VERSION);

```

As expected this returns ``Hello, 8.1.2!`` on my local machine.
And here is what I get on the server:
![image](https://user-images.githubusercontent.com/6942070/155229491-d159c976-3150-4ad4-a255-930c42e52d1a.png)

This is the exact same error I get when running following code on my local machine:
```php
<?php

namespace Test;

function greeter(string $name) {
  echo "Hello, ${name}!";
}

$mycallable = greeter();

$mycallable(PHP_VERSION);

```

**Comment by Girgias** [comment]
You do not have the ``...`` in the definition of ``$mycallable`` in ``greeter()`` in your second example.

**Comment by L1Q** [comment]
@Girgias exactly! That's what makes me think the server version is not working properly. not providing ``...`` means calling the function immediately. 
Interestingly enough, when providing full namespace when creating the closure, it runs on the server.
This code works fine on both machines:
```php
<?php

namespace Test;

function greeter(string $name) {
  echo "Hello, ${name}!";
}

$mycallable = \Test\greeter(...);

$mycallable(PHP_VERSION);


```

**Comment by L1Q** [comment]
Here, this fails again with ``ArgumentCountError``:
```php
<?php

namespace Test;

$mycallable = \Test\greeter(...);

function greeter(string $name) {
  echo "Hello, ${name}!";
}

$mycallable(PHP_VERSION);

```

**Comment by Girgias** [comment]
This runs just fine on https://3v4l.org/vS7IB are you running with JIT or Opcache?

**Comment by L1Q** [comment]
> are you running with JIT or Opcache?

YES! Server is running with ``opcache.enable``
